File: ir/src/main/java/org/ksharp/ir/truffle/IntegerNode.java
package org.ksharp.ir.truffle;

import com.oracle.truffle.api.frame.VirtualFrame;

public class IntegerNode extends KSharpNode {
    private final long value;

    public IntegerNode(long value) {
        this.value = value;
    }

    @Override
    public Object execute(VirtualFrame frame) {
        return value;
    }
}

File: ir/src/main/kotlin/org/ksharp/ir/truffle/IntegerNode.kt
package org.ksharp.ir.truffle

import com.oracle.truffle.api.frame.VirtualFrame
import java.math.BigInteger

abstract class IntegerNode : KSharpNode() {

    abstract fun executeValue(frame: VirtualFrame): Long

    abstract fun executeBigIntValue(frame: VirtualFrame): BigInteger
    
}
